coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,300 @@
+        (0, 1, 2, 3), default=2)
+    enforce_newline_at_EOF = Taste[bool](
+        'Checks if a file ends with a single newline.',
+        (True, False), default=True)
+
+
+@Spacing.subaspect
+class SpacesAroundOperator:
+    """
+    This aspect checks on the spacing around operators in your codebase.
+    """
+    class docs:
+        example = """
+        def f(a):
+            return 37+-+a[42-x :  y**3]  #Missing spaces around operators.
+        """
+        example_language = 'python'
+        importance_reason = """
+        Improves the readability and maintainability of your code.
+        """
+        fix_suggestions = """
+        """
+    spaces_around_operators = Taste[int](
+        'Represents the number of space to be used around operators.',
+        (0, 1), default=1)
+    spaces_before_colon = Taste[int](
+        'Represents the number of blank spaces before colons.',
+        (0, 1), default=0)
+    spaces_after_colon = Taste[int](
+        'Represents the number of blank spaces after colons.',
+        (0, 1), default=1)
+
+
+@Spacing.subaspect
+class Indentation:
+    """
+    This aspect checks on indentation in your file.
+    """
+    class docs:
+        example = """
+        # here is a example of code with some indentation issues
+
+        def func(a, b):
+        ...result = a**b  # Use 4 spaces per indentation level
+        \\treturn result  # Use either spaces or tabs for indentation
+        """
+        example_language = 'python'
+        importance_reason = """
+        Improves the readability of your code.
+        """
+        fix_suggestions = """
+        Use either spaces or tabs for indentation not both. In case you are
+        using spaces, specified the number of space to be use per indentation
+        level.
+        """
+    indent_size = Taste[int](
+        'Represents the number of space per indentation level.',
+        (4, 8), default=4)
+    use_spaces = Taste[bool](
+        'Determines whether spaces should be use for indentation',
+        (True, False), default=True)
+
+
+@Formatting.subaspect
+class Quotation:
+    """
+    This aspect checks on the quotation mark used in your code.
+    """
+    class docs:
+        example = """
+        # Here is an example of code where both '' and "" quotation mark
+        # Are used.
+
+        string = 'coala is always written with lowercase c.'
+        string = "coala is always written with lowercase c."
+        """
+        example_language = 'python'
+        importance_reason = """
+        Using the same quotation (if possible) in your code, improve on its
+        readability by introducing consistency.
+        """
+        fix_suggestions = """
+        Choosing a preferred quotation and use everywhere (if possible).
+        """
+    preferred_quotation = Taste[str](
+        'Represents the preferred quotation',
+        ('\'', '"'), default='\'')
```

Comment:
@Asnelchristian Even if `preferred_quotation` is `'\''` you can make exceptions to avoid escaping, like we do in `coala` itself, so allowing `"'"` for example

@sils